### ISSUE TITLE
[SC2116] exception: `$(echo ...)` used to compose regular expression patterns

### ISSUE THREAD

**Issue by saveriomiroddi** [koalaman/shellcheck#1746]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash

[[ 123 =~ $(echo '^123\b') ]] && echo match
```

#### Here's what shellcheck currently says:

```
Line 3:
[[ 123 =~ $(echo '^123\b') ]] && echo match
          ^-- SC2116: Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.
```

#### Here's what I wanted or expected to see:

No warnings.

Since non-trivial regular expressions (to be matched with the `=~` operator) can't be written inline, one needs a workaround, typically `$(echo <expression>)`, which ideally should not raise a warning.

I'm also a bit perplexed about Shellcheck recognizing the `=~` operator as a command, which causes the warning to be raises (however, I'm not familiar with the internals).

**Comment by mk-pmb** [comment]
I usually assign them to a variable.

```bash
RGX='^\ ([a-z])..'; [[ ' hallo ' =~ $RGX ]]; echo "$?:[${BASH_REMATCH[*]}]"
```
